src/bsp/src/stm32/src/interloc/src/InterlocTimeManager.cpp

```diff
@@ -17,73 +62,124 @@ void InterlocTimeManager::setSlodId(uint16_t slotId) {
 }
 
 void InterlocTimeManager::updateTimings() {
-    volatile uint64_t pollRxToResponseTxOffset =
-        POLL_AIR_TIME_WITH_PREAMBLE_US + POLL_TO_FIRST_RESPONSE_GUARD_US +
-        (uint64_t)(m_slotId - 1) *
-            (uint64_t)(RESPONSE_AIR_TIME_WITH_PREAMBLE_US + RESPONSE_TO_RESPONSE_GUARD_US);
+    // compute constant values air time with their preamble
+    m_pollAirTimeWithPreambleUs = computeAirTimeWithPreambleUs(sizeof(UWBMessages::TWRPoll) << 3);
+    m_responseAirTimeWithPreambleUs =
+        computeAirTimeWithPreambleUs(sizeof(UWBMessages::TWRResponse) << 3);
+    m_finalAirTimeWithPreambleUs = computeAirTimeWithPreambleUs(sizeof(UWBMessages::TWRFinal) << 3);
+    m_pollToFirstResponseGuardUs = getReadWriteSPITimeUs(sizeof(UWBMessages::TWRPoll)) +
+                                   POLL_PROCESSING_GUARD + getPreambleAirTimeUs();
+}
 
-    volatile uint64_t pollTxToFinalTxOffset =
-        POLL_AIR_TIME_WITH_PREAMBLE_US + POLL_TO_FIRST_RESPONSE_GUARD_US +
-        m_numSlots * (RESPONSE_AIR_TIME_WITH_PREAMBLE_US + RESPONSE_TO_RESPONSE_GUARD_US);
+constexpr float getShrSymbolDurationUs() {
+    // see table 13 in datasheet
+    switch (PRF_SPEED) {
+    case 16:
+        return 993.59 / 1000;
+    default:
+        return 1017.63 / 1000;
+    }
+}
 
-    volatile uint64_t pollRxToFinalRxOffset = pollTxToFinalTxOffset - RX_BEFORE_TX_GUARD_US;
+constexpr float getPhrSymbolDurationUs() {
+    // see table 13 in datasheet
+    switch (DECAWAVE_TX_RATE_HZ) {
+    case 110000:
+        return 8205.13 / 1000;
+    default:
+        return 1025.64 / 1000;
+    }
+}
 
-    volatile uint64_t slotToSlotOffsetUs =
-        pollTxToFinalTxOffset + FINAL_AIR_TIME_WITH_PREAMBLE_US + FINAL_TO_POLL_GUARD_US;
+uint16_t InterlocTimeManager::getPreambleAirTimeUs() {
+    float shrBitLength = PREAMBULE_SEQUENCE_LENGTH + START_FRAME_DELIMITER_LENGTH;
+    float phrBitLength = PHY_HEADER_LENGTH;
 
-    for (unsigned int slotIdx = 1; slotIdx <= m_numSlots; slotIdx++) {
-        m_RespRxStartOffset[slotIdx - 1] =
-            UUS_TO_DWT_TIME *
-            ((slotIdx - 1) * (RESPONSE_AIR_TIME_WITH_PREAMBLE_US + RESPONSE_TO_RESPONSE_GUARD_US) +
-             POLL_TO_FIRST_RESPONSE_GUARD_US - RX_BEFORE_TX_GUARD_US);
-    }
+    return (uint16_t)(shrBitLength * getShrSymbolDurationUs() +
+                      phrBitLength * getPhrSymbolDurationUs());
+}
 
-    m_pollRxToResponseTxOffsetDTU = pollRxToResponseTxOffset * UUS_TO_DWT_TIME;
-    m_pollTxToFinalTxOffsetDTU = pollTxToFinalTxOffset * UUS_TO_DWT_TIME;
-    m_pollRxToFinalRxOffsetDTU = pollRxToFinalRxOffset * UUS_TO_DWT_TIME;
-    m_slotToSlotOffsetDTU = slotToSlotOffsetUs * UUS_TO_DWT_TIME;
+uint16_t InterlocTimeManager::getReadWriteSPITimeUs(uint16_t bitLength) {
+    return bitLength * 1000000 / SPI_SPEED_HZ;
```

Comment:
Maybe not now, but if we want to be more precise, we could add a constant overhead that is different for read/write (depending on number of registers we need to read/write to)